### PR TITLE
Tr5 hk gun with combineable lasersight

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**

### CODE REVIEW

File: TombEngine/Game/gui.cpp [hunk 1]
```diff
@@ -2381,6 +2398,23 @@ namespace TEN::Gui
...
 
+		if (lara->Weapons[(int)LaraWeaponType::HK].Present)
+		{
+			lara->Weapons[(int)LaraWeaponType::HK].Weaponmode = LaraWeaponTypeCarried::WTYPE_AMMO_1;
+			lara->Weapons[(int)LaraWeaponType::HK].SelectedAmmo = WeaponAmmoType::Ammo1;
+
+			if (Ammo.CurrentHKAmmoType == 1)
+			{
+				lara->Weapons[(int)LaraWeaponType::HK].Weaponmode = LaraWeaponTypeCarried::WTYPE_AMMO_2;
+				lara->Weapons[(int)LaraWeaponType::HK].SelectedAmmo = WeaponAmmoType::Ammo1;
```
Comment (Lwmte) [inline TombEngine/Game/gui.cpp:2409]: Is this intended that `SelectedAmmo` is also `Ammo1` here, or it is copypaste error?

File: TombEngine/Game/gui.cpp [hunk 2]
```diff
@@ -2381,6 +2398,23 @@ namespace TEN::Gui
...
+			}
+			else if (Ammo.CurrentHKAmmoType == 2)
+			{
+				lara->Weapons[(int)LaraWeaponType::HK].Weaponmode = LaraWeaponTypeCarried::WTYPE_AMMO_3;
+				lara->Weapons[(int)LaraWeaponType::HK].SelectedAmmo = WeaponAmmoType::Ammo1;
```
Comment (Lwmte) [inline TombEngine/Game/gui.cpp:2414]: Same as above

File: TombEngine/Game/Lara/lara_one_gun.cpp [hunk 3]
```diff
@@ -146,34 +155,45 @@ void AnimateShotgun(ItemInfo* laraItem, LaraWeaponType weaponType)
...
 					item->Animation.TargetState = WEAPON_STATE_AIM;
 			}
 
-
-			if (item->Animation.TargetState != WEAPON_STATE_RECOIL && 
-				weaponType == LaraWeaponType::HK &&
-				!lara->Weapons[(int)LaraWeaponType::HK].HasSilencer)
+			if (item->Animation.TargetState != WEAPON_STATE_RECOIL &&
+				HK.Flag &&
+				weaponType == LaraWeaponType::HK )
```
Comment (Lwmte) [inline TombEngine/Game/Lara/lara_one_gun.cpp:185]: Extra unneeded space here before closing bracket.

File: TombEngine/Game/Lara/lara_one_gun.cpp [hunk 4]
```diff
@@ -1632,12 +1651,25 @@ void CrossbowBoltControl(short itemNumber)
...
-
+	auto& weapon = lara->Weapons[(int)LaraWeaponType::HK];
 	auto angles = EulerAngles(
 		lara->LeftArm.Orientation.x,
 		lara->LeftArm.Orientation.y + laraItem->Pose.Orientation.y,
 		0
 	);
+	
+	if (weapon.Weaponmode == LaraWeaponTypeCarried::WTYPE_AMMO_1)
+		weapon.Interval = 12;
```
[the remark this answers is not in the document]
Comment (Nemoel-Tomo) [reply TombEngine/Game/Lara/lara_one_gun.cpp]: fixed

File: TombEngine/Game/gui.cpp (hunk 2 above)
Comment (Nemoel-Tomo) [reply TombEngine/Game/gui.cpp:2414]: see above

File: TombEngine/Game/Lara/lara_one_gun.cpp (hunk 3 above)
Comment (Nemoel-Tomo) [reply TombEngine/Game/Lara/lara_one_gun.cpp:186]: fixed

**Pull request merged**